src/middleware/upload.ts

```diff
@@ -0,0 +1,62 @@
+import { Request } from "express";
+import { fileTypeFromBuffer } from "file-type";
+import { lookup } from "mime-types";
+import multer from "multer";
+
+// Allowed file types
+const ALLOWED_MIMES = new Set([
+  "image/jpeg",
+  "image/png",
+  "image/gif",
+  "application/pdf",
+  "text/plain",
+  "application/json",
+]);
+
+// File validation error types
+export enum FileValidationError {
```

Comment:
Just wanted to make you aware of the idiosyncrasies of enums: https://www.youtube.com/watch?v=jjMbPt_H3RQ